File: app/src/main/java/igrek/todotree/ui/tree/TreeItemTouchListener.java
package igrek.todotree.ui.tree;

import android.view.MotionEvent;
import android.view.View;

class TreeItemTouchListener implements View.OnTouchListener {
	
	private TreeListView listView;
	private int position;
	
	TreeItemTouchListener(TreeListView listView, int position) {
		this.listView = listView;
		this.position = position;
	}
	
	@Override
	public boolean onTouch(View v, MotionEvent event) {
		switch (event.getAction() & MotionEvent.ACTION_MASK) {
			case MotionEvent.ACTION_DOWN:
				listView.onItemTouchDown(position, event, v);
				break;
		}
		return false;
	}
}

File: app/src/main/java/igrek/todotree/ui/treelist/TreeItemTouchListener.kt
package igrek.todotree.ui.treelist

import android.view.MotionEvent
import android.view.View
import android.view.View.OnTouchListener

internal class TreeItemTouchListener(
    private val listView: TreeListView,
    private val position: Int
) : OnTouchListener {
    override fun onTouch(v: View, event: MotionEvent): Boolean {
        when (event.action and MotionEvent.ACTION_MASK) {
            MotionEvent.ACTION_DOWN -> listView.onItemTouchDown(
                position, event, v
            )
        }
        return false
    }
}